starter/slack-spring-boot-autoconfigure/src/main/kotlin/com/kreait/slack/broker/autoconfiguration/UserStoreAutoConfiguration.kt

```diff
@@ -0,0 +1,72 @@
+package com.kreait.slack.broker.autoconfiguration
+
+import com.kreait.slack.api.SlackClient
+import com.kreait.slack.broker.store.user.FileUserStore
+import com.kreait.slack.broker.store.user.InMemoryUserStore
+import com.kreait.slack.broker.store.user.UserChangedEventReceiver
+import com.kreait.slack.broker.store.user.UserInstallationReceiver
+import com.kreait.slack.broker.store.user.UserJoinedEventReceiver
+import com.kreait.slack.broker.store.user.UserStore
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+
+/**
+ * AutoConfiguration that registers a [UserStore]
+ */
+@EnableConfigurationProperties(SlackBrokerConfigurationProperties::class)
+@Configuration
+open class UserStoreAutoConfiguration : WebMvcConfigurer {
+
+    /**
+     * Registers the [InMemoryUserStore] if no other is defined and the property [SlackBrokerConfigurationProperties.USER_STORE].type is set to memory
+     */
+    @ConditionalOnProperty(prefix = SlackBrokerConfigurationProperties.USER_STORE, name = ["type"], havingValue = "memory")
+    @ConditionalOnMissingBean
+    @Bean
+    open fun userStore(): UserStore {
+        return InMemoryUserStore()
+    }
+
+    /**
+     * Registers the [FileUserStore] if no other is defined and the property [SlackBrokerConfigurationProperties.USER_STORE].type is set to file
+     */
+    @ConditionalOnProperty(prefix = SlackBrokerConfigurationProperties.USER_STORE, name = ["type"], havingValue = "file")
+    @ConditionalOnMissingBean
+    @Bean
+    open fun localUserStore(): UserStore {
+        return FileUserStore()
+    }
+
+    /**
+     * Registers the [UserInstallationReceiver] if a [UserStore] registered, that downloads the user after an installation
+     */
+    @ConditionalOnBean(UserStore::class)
+    @Bean
+    open fun userManager(applicationContext: ApplicationContext, slackClient: SlackClient, userStore: UserStore): UserInstallationReceiver? {
```

Comment:
rename this method to userInstallationReceiver() and remove the applicationContext as Parameter. and the returnType can not be optional